crates/biome_js_analyze/src/lint/nursery/no_enum.rs

```diff
@@ -0,0 +1,66 @@
+use biome_analyze::{context::RuleContext, declare_lint_rule, Ast, Rule, RuleDiagnostic};
+use biome_console::markup;
+use biome_js_syntax::TsEnumDeclaration;
+use biome_rowan::AstNode;
+
+declare_lint_rule! {
+    /// Disallow TypeScript enum.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```ts,expect_diagnostic
+    /// enum Foo {
+    ///     BAR = 'bar'
+    ///     BAZ = 'baz'
+    /// }
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```ts
+    /// const Foo {
+    ///     BAR: 'bar',
+    ///     BAZ: 'baz',
+    /// }
+    /// ```
+    ///
+    /// ```ts
+    /// type Foo = 'bar' | 'baz'
+    /// ```
+    ///
+    ///
+    pub NoEnum {
+        version: "next",
+        name: "noEnum",
+        language: "ts",
+        recommended: false,
+    }
+}
+
+impl Rule for NoEnum {
+    type Query = Ast<TsEnumDeclaration>;
+    type State = ();
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(_ctx: &RuleContext<Self>) -> Self::Signals {
+        Some(())
+    }
+
+    fn diagnostic(ctx: &RuleContext<Self>, _state: &Self::State) -> Option<RuleDiagnostic> {
+        Some(
+            RuleDiagnostic::new(
+                rule_category!(),
+                ctx.query().range(),
+                markup! {
+                    "Don't use "<Emphasis>"enum"</Emphasis>
+                },
+            )
+            .note(markup! {
+                "Prefer using JavaScript objects or TypeScript unions over enums."
+            }),
```

Comment:
The diagnostics are good, however we miss the explanation of the error. Here, we need to explain why a user shouldn't use enums.

Please refer to our rule pillars: https://biomejs.dev/linter/#rule-pillars

Reply:
Hey @nickfla1, I noticed [`5694370` (#3653)](https://github.com/biomejs/biome/pull/3653/commits/5694370ef7bf19a1d08574e3e1de7b44ecc2f453). The reason should be added **to the diagnostic** too :)